### ISSUE TITLE
[Feature] Enable mouse zooming and translating on maps even when the mouse is not over the map itself

### ISSUE THREAD

**Issue by jgoncab-arte** [apache/echarts#20444]
### What problem does this feature solve?

It's a bit confusing that you can only zoom with the mouse when it's over the map. On some maps with too much space between regions you can't zoom with the mouse in the middle. On the other hand, on any map (like https://echarts.apache.org/examples/en/editor.html?c=map-usa) if you zoom out enough, the map becomes very small and there's not much room left to zoom again:
![image](https://github.com/user-attachments/assets/ed797de8-e5de-4600-9599-dc29f78f02c0)
Additionally, it appears that regions with no associated data are also ignored when zooming the map.

### What does the proposed API look like?

Probably the API shouldn't change, but in case someone prefers to keep the current behavior maybe something like:
`{ roam: true, roamOn: ("everywhere" | "map")}`

**Comment by helgasoft** [comment]
Chart size should not get out of control. There is [scaleLimit.min](https://echarts.apache.org/en/option.html#series-map.scaleLimit.min) for that.
There is also toolbox Restore button.
Is a new feature really needed ?

**Comment by jgoncab-arte** [comment]
Of course, users have other options, but I found it confusing the first time until I realized that the problem was that the mouse was not over the map. Perhaps the case shown in the image is not the most concerning, but I have a world map where I cannot zoom if the mouse is over the Atlantic Ocean.

I tried to solve it with events but the performance is clearly worse. It's probably not the most urgent thing on your plate, but it would be nice to have.

**Comment by helgasoft** [comment]
> I have a world map where I cannot zoom if the mouse is over the Atlantic Ocean.

Could you share it? Same ocean works fine in [official example](https://echarts.apache.org/examples/en/editor.html?c=map-usa).
